### PR TITLE
refactor: remove Node.js and pull streams

### PR DESCRIPTION
**Pull request by alanshaw**
Also changes `stat` and `ls` to return CID's not strings from core and removes the `long` option from `ls` in core (is applied by CLI/HTTP API).

refs https://github.com/ipfs/interface-js-ipfs-core/issues/394

BREAKING CHANGE: `files.readPullStream` and `files.readReadableStream` have been removed.

BREAKING CHANGE: `files.read` now returns an async iterable.

BREAKING CHANGE: `files.lsPullStream` and `files.lsReadableStream` have been removed.

BREAKING CHANGE: `files.ls` now returns an async iterable.

BREAKING CHANGE: `files.ls` results now contain a `cid` property (whose value is a [CID instance](https://github.com/multiformats/js-cid)) instead of a `hash` property.

BREAKING CHANGE: `files.ls` no longer takes a `long` option (in core) - you will receive all data by default.

BREAKING CHANGE: `files.stat` result now contains a `cid` property (whose value is a [CID instance](https://github.com/multiformats/js-cid)) instead of a `hash` property.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,23 +52,23 @@
...
     "hamt-sharding": "~0.0.2",
-    "interface-datastore": "~0.7.0",
+    "interface-datastore": "^0.8.0",
     "ipfs-multipart": "~0.2.0",
-    "ipfs-unixfs": "~0.1.16",
-    "ipfs-unixfs-exporter": "~0.38.0",
-    "ipfs-unixfs-importer": "~0.40.0",
+    "ipfs-unixfs": "^0.2.0",
+    "ipfs-unixfs-exporter": "^0.39.0",
+    "ipfs-unixfs-importer": "^0.42.0",
```
Comment (achingbrain) [inline package.json]: Do you need the importer/exporter/unixfs with metadata support for this?

**Review by achingbrain — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (achingbrain) [reply package.json]: I ask because I wouldn't pull them in until the dust has settled on https://github.com/ipfs/specs/pull/229

Comment (alanshaw) [reply package.json]: No I do not

Comment (alanshaw) [reply package.json]: I was just like, only forwards...

Comment (achingbrain) [reply package.json]: A noble approach. Bravo, sir!

I'd, uh, still back the upgrade out of this PR though..

Comment (alanshaw) [reply package.json]: Not too far forwards!

**Review by achingbrain — APPROVED** [review]

**Pull request merged**